src/main/resources/org/jenkinsci/plugins/workflow/job/console/NewNodeConsoleNote/script.js

```diff
@@ -1,62 +1,79 @@
+ */
+setTimeout(() => {
+    /**
+     * Specifies something to do when an element matching a CSS selector is encountered.
+     *
+     * @param {String} selector a CSS selector triggering your behavior
+     * @param {String} id combined with selector, uniquely identifies this behavior; prevents duplicate registrations
+     * @param {Number} priority relative position of this behavior in case multiple apply to a given element; lower numbers applied first (sorted by id then selector in case of tie); choose 0 if you do not care
+     * @param {Function} behavior callback function taking one parameter, a (DOM) {@link Element}, and returning void
+     */
+    Behaviour.specify("span.pipeline-new-node", 'NewNodeConsoleNote', 0, function (e) {
+        var nodeId = e.getAttribute('nodeId')
+        var oid = e.getAttribute('nodeId')
+        var nodes = $$('.pipeline-new-node')
+        var enclosings = new Map() // id → enclosingId
+        var labels = new Map() // id → label
+
+        if (e.processedNewNodeConsoleNote) {
+            return
         }
-    }
-    var nodeId = e.getAttribute('nodeId')
-    var startId = e.getAttribute('startId')
-    if (startId == null || startId == nodeId) {
-        e.innerHTML = e.innerHTML.replace(/.+/, '$&<span class="pipeline-show-hide"> (<a href="#" onclick="showHidePipelineSection(this); return false">hide</a>)</span>')
-        // TODO automatically hide second and subsequent branches: namely, in case a node has the same parent as an earlier one
-    }
-    // The CSS rule for branch names only needs to be added once per node, so we
-    // check in case we are viewing the truncated log and have already processed
-    // a duplicate synthetic span element for this node.
-    var maybeDupeNodes = $$('[nodeid=\"'+nodeId+'\"].pipeline-new-node');
-    for (var i = 0; i < maybeDupeNodes.length; i++) {
-        var node = maybeDupeNodes[i];
-        if (node !== e && node.processedNewNodeConsoleNote) {
-            return;
+        e.processedNewNodeConsoleNote = true
+
+        var label = e.getAttribute('label')
+        if (label != null) {
+            var html = e.innerHTML
+            var suffix = ' (' + label.escapeHTML() + ')';
+            if (html.indexOf(suffix) < 0) {
+                e.innerHTML = e.innerHTML.replace(/.+/, '$&' + suffix) // insert before EOL
+            }
         }
-    }
-    var nodes = $$('.pipeline-new-node')
-    var enclosings = new Map() // id → enclosingId
-    var labels = new Map() // id → label
-    for (var i = 0; i < nodes.length; i++) {
-        var node = nodes[i]
-        var oid = node.getAttribute('nodeId')
-        enclosings.set(oid, node.getAttribute('enclosingId'))
-        labels.set(oid, node.getAttribute('label'))
-    }
-    var id = nodeId
-    while (true) {
-        id = enclosings.get(id)
-        if (id == null) {
-            break
+
+        // The CSS rule for branch names only needs to be added once per node, so we
```

Comment:
BTW https://github.com/jenkinsci/workflow-job-plugin/pull/164/files?diff=unified&w=1 improves reviewability due to indentation changes.